### DISCUSSION TITLE
refiner with SD1.5

### DISCUSSION THREAD

**Discussion by vidka001** [vladmandic/sdnext#2359] [Q&A]
Hi! I noticed interesting thing using refiner with SD1.5 checkpoints: when second pass started, generated image changed completely. I tried set "Refiner start" to 0,95, but it didn’t help.
In Automatic1111 refiner just adds some details to generated image. I don’t know how to reproduce that behavior in SD.Next.

**Comment by Tillerz** [comment]
Hi, I just tried that too. It first renders the same image at higher resolution, but then throws an error in the log
`08:24:31-717415 ERROR    Prompt parser encode: "LayerNormKernelImpl" not implemented for 'Half'`
and bugs out with showing a completely different image. Do you also get an error in the log or does it go through fine?

**Comment by vidka001** [comment]
Yes, I have the same error:
ERROR    Prompt parser encode: "LayerNormKernelImpl" not implemented for 'Half'

**Comment by vladmandic** [comment]
which backend, gpu, precision? also run `webui --debug` and post little bit more before this one line error.

**Comment by vidka001** [comment]
System Info

Pipeline: diffusers

GPU:
device: NVIDIA GeForce RTX 4090 (1) (sm_90) (8, 9)
cuda: 12.1
cudnn: 8801
device: NVIDIA GeForce RTX 4090 (1) (sm_90) (8, 9)
cuda: 12.1
cudnn: 8801
driver: 531.79

Device Info:
active: cuda
dtype: torch.float16
vae: torch.float16
unet: torch.float16

Torch: 
2.1.0+cu121 Autocast  half

Libs:
xformers: 
diffusers: 0.21.4
transformers: 4.30.2

Cross-attention:
Scaled-Dot-Product


> 
<details><summary>Detailed log</summary>
<p>

10:50:43-905541 DEBUG    Desired Torch parameters: dtype=FP16 no-half=False no-half-vae=False upscast=False
10:50:43-906541 INFO     Setting Torch parameters: device=cuda dtype=torch.float16 vae=torch.float16 unet=torch.float16 context=no_grad fp16=True bf16=False
10:50:43-907541 INFO     Autodetect: refiner="Stable Diffusion" class=StableDiffusionPipeline file="D:\sd\automatic\models\Stable-diffusion\epicrealism_naturalSin.safetensors" size=2034MB
10:50:45-438323 DEBUG    Setting refiner: pipeline=StableDiffusionPipeline config={'low_cpu_mem_usage': True, 'torch_dtype': torch.float16, 'load_connected_pipeline': True, 'variant': 'fp16', 'extract_ema': True,
                         'force_zeros_for_empty_prompt ': True, 'requires_aesthetics_score': False, 'use_safetensors': True}
10:50:45-440324 DEBUG    Setting refiner: enable VAE slicing
10:50:45-441324 DEBUG    Setting refiner: enable VAE tiling
10:50:45-444323 DEBUG    Setting refiner VAE: name=None upcast=True
10:50:45-445323 DEBUG    Moving refiner model to CPU
10:50:45-455872 INFO     Loaded embeddings: loaded=2 skipped=0 time=0.00s
10:50:45-743071 DEBUG    gc: collected=1269 device=cuda {'ram': {'used': 5.43, 'total': 63.15}, 'gpu': {'used': 3.71, 'total': 23.99}, 'retries': 0, 'oom': 0}
10:50:45-748071 INFO     Loaded refiner: time=1.73s { load=1.73s } native=512 {'ram': {'used': 5.43, 'total': 63.15}, 'gpu': {'used': 3.71, 'total': 23.99}, 'retries': 0, 'oom': 0}
10:50:45-823071 DEBUG    Sampler: sampler=DPM SDE config={'num_train_timesteps': 1000, 'beta_start': 0.00085, 'beta_end': 0.012, 'beta_schedule': 'scaled_linear', 'prediction_type': 'epsilon', 'use_karras_sigmas': True}
10:50:46-232072 DEBUG    Diffuser pipeline: StableDiffusionPipeline task=DiffusersTaskType.TEXT_2_IMAGE set={'prompt_embeds': torch.Size([1, 154, 768]), 'negative_prompt_embeds': torch.Size([1, 154, 768]), 'guidance_scale': 6,
                         'generator': device(type='cuda'), 'output_type': 'latent', 'num_inference_steps': 20, 'eta': 1.0, 'guidance_rescale': 0.7, 'height': 768, 'width': 512, 'parser': 'Full parser'}
Progress  2.48it/s █████                               15% 3/20 00:01 00:06 Base10:50:47-801453 DEBUG    Loaded VAE decode approximate: model="models\VAE-approx\model.pt"

Progress  6.43it/s █████████████████████████████████ 100% 20/20 00:03 00:00 Base
10:50:49-349645 DEBUG    Moving to CPU: model=UNet
10:50:50-830962 DEBUG    VAE decode: name=baked dtype=torch.float16 upcast=False images=1 latents=torch.Size([1, 4, 96, 64]) time=1.48s
10:50:50-846961 DEBUG    Saving: image="outputs\text\00207-animerge_v22Yoru-beautiful landscape-before-refiner.jpg" type=JPEG size=512x768
10:50:50-862348 DEBUG    Moving to CPU: model=base
10:50:51-172352 DEBUG    Sampler: sampler=DPM SDE config={'num_train_timesteps': 1000, 'beta_start': 0.00085, 'beta_end': 0.012, 'beta_schedule': 'scaled_linear', 'prediction_type': 'epsilon', 'use_karras_sigmas': True}
10:50:51-742544 ERROR    Prompt parser encode: "LayerNormKernelImpl" not implemented for 'Half'
10:50:51-743545 DEBUG    Diffuser pipeline: StableDiffusionPipeline task=DiffusersTaskType.TEXT_2_IMAGE set={'prompt': 1, 'negative_prompt': 1, 'guidance_scale': 6, 'generator': device(type='cuda'), 'output_type': 'latent',
                         'num_inference_steps': 37, 'eta': 1.0, 'guidance_rescale': 0.7, 'height': 768, 'width': 512, 'parser': 'Fixed attention'}
Progress 10.05it/s ██████████████████████████████ 100% 37/37 00:03 00:00 Refiner
10:50:55-450092 DEBUG    Moving to CPU: model=UNet
10:50:55-719487 DEBUG    VAE decode: name=baked dtype=torch.float16 upcast=False images=1 latents=torch.Size([1, 4, 96, 64]) time=0.269s
10:50:55-723486 DEBUG    Moving to CPU: model=refiner
10:50:56-316473 DEBUG    Saving: image="outputs\text\00208-animerge_v22Yoru-beautiful landscape.jpg" type=JPEG size=512x768
10:50:56-405477 INFO     Processed: images=1 time=12.75s its=1.57 memory={'ram': {'used': 6.18, 'total': 63.15}, 'gpu': {'used': 2.17, 'total': 23.99}, 'retries': 0, 'oom': 0}

</p>
</details>

**Comment by vladmandic** [comment]
just pushed an update, should help with this.

**Comment by vidka001** [comment]
Thanks! Refined image is much similar to base image now.
Is refiner_start parameter ignored for sd15?

**Reply by vladmandic** [reply]
Yes, that param is ignored for SD15
